zipkin-server/src/main/java/zipkin/server/ZipkinServerConfiguration.java

```diff
@@ -46,8 +48,14 @@ CollectorSampler traceIdSampler(@Value("${zipkin.collector.sample-rate:1.0}") fl
 
   @Bean
   @ConditionalOnMissingBean(CollectorMetrics.class)
-  CollectorMetrics metrics(CounterService counterService, GaugeService gaugeService) {
-    return new ActuateCollectorMetrics(counterService, gaugeService);
+  CollectorMetrics metrics(Optional<CounterBuffers> counterBuffers, Optional<GaugeBuffers> gaugeBuffers) {
```

Comment:
this is new logic, with two conditionals and a lot of commentary. we try not to rely exclusively on integration tests for things that can be unit tested because it makes bugs harder to find later. We have been trying to be better at tests, lately, adding unit tests for the classes we change, especially auto-configuration as bugs and drift are easy there.

If it isn't clear how to setup a test like this (ex [this link](https://github.com/openzipkin/zipkin-aws/blob/master/autoconfigure/sparkstreaming-stream-kinesis/src/test/java/zipkin/sparkstreaming/autoconfigure/stream/kinesis/ZipkinKinesisStreamFactoryAutoConfigurationTest.java) doesn't help). That's cool, I can try and help. Just asking if you can.